app/queues/notifications/expiration-notification.server.ts

```diff
@@ -0,0 +1,160 @@
+// Third party modules
+import { Worker, Queue, UnrecoverableError } from 'bullmq';
+
+// Internal modules
+import { redis } from '~/lib/redis.server';
+import logger from '~/lib/logger.server';
+import { addNotification } from './notifications.server';
+import { prisma } from '~/db.server';
+
+// Types
+import type { NotificationData } from './notifications.server';
+
+export enum RecordType {
+  Certificate = 'certificate',
+  DnsRecord = 'record',
+}
+export interface ExpirationStatusPayload {
+  type: RecordType;
+}
+
+// name for the queue
+const expirationNotificationQueueName = 'expiration-notification';
+
+const updateNotificationStatus = (type: RecordType, id: number) => {
+  // throw if any of the arguments are missing
+  if (!type || !id) {
```

Comment:
I'm confused about the need to include these.  The types you define in 24 don't allow them to be null/undefined.  Can you clarify why you do this extra check?

Reply:
That's an error. 

To clarify: if I infer the type, there is no need to check because the value can't be null or undefined, as Typescript won't allow it.